geo-types/src/geometry/coord.rs

```diff
@@ -268,54 +265,57 @@ impl<T: CoordNum> Zero for Coord<T> {
 }
 
 #[cfg(any(feature = "approx", test))]
-impl<T: CoordNum + AbsDiffEq> AbsDiffEq for Coord<T>
-where
-    T::Epsilon: Copy,
-{
-    type Epsilon = T::Epsilon;
-
-    #[inline]
-    fn default_epsilon() -> T::Epsilon {
-        T::default_epsilon()
-    }
+mod approx_integration {
+    use super::*;
+    use approx::{AbsDiffEq, RelativeEq, UlpsEq};
+
+    impl<T> AbsDiffEq for Coord<T>
+    where
```

Comment:
Other than the new UlpsEq impl below, these changes are formatting changes.

Though I also simplified/standardized some of our type constraints.

e.g. here `T::Epsilon: Copy` is replaced by `AbsDiffEq<Epsilon=T>`

Since `T: CoordNum`, we already know it's `Copy`.

In **theory**, the previous implementation might have allowed something like:

`Coord<f32>.abs_diff(Coord<f64>)`

But because `pub trait AbsDiffEq<Rhs = Self>` we only ever supported comparing to `Self` anyway (and I think that's just fine).

Bottom line, this doesn't change behavior and is more succinct/uniform with the other declarations.

Reply:
BTW I recommend turning on the option to hide whitespace changes for this diff.